### PR TITLE
HEART FAILURE IS MORE DEADLIERY! also coffee now restores stamina but can be addictive, is deadly if overdosed

### PR DESCRIPTION
**Pull request by Sishen1542**
## About The Pull Request
Robusto brand space coffee, perfect for the spaceman who needs that tiny extra kick in the rear!
Though it IS strong, so make sure to watch your caffeine intake!

Coffee now restores stamina, is addictive at 30 units and will overdose at 61.
Addiction deals stamina loss & shortness of breath periodically.
Overdosing will make you develop heart failure, randomly move you a few tiles and make you drop items randomly.

Also, made heart failure stage up faster to be more of a threat.

## Why It's Good For The Game
Stamina man is bad, adding a way to (slightly) tilt things in your favor shouldn't hurt too much. Very punishing if abused/chugged.
Heart failure was laughably slow previously, I've buffed the values to be punishing to people who abuse the coffee.
People who fall victim to random heart attacks might want to call out for help faster, though.

Up for value tweaks/testmerging.

## Changelog
:cl:
balance: rebalanced something
/:cl:

### CODE REVIEW

**Review by deathride58 — DISMISSED** [review]

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 1]
```diff
@@ -262,20 +262,61 @@
...
+	addiction_threshold = 30
+	overdose_threshold = 61
 	taste_description = "bitterness"
 	glass_icon_state = "glass_brown"
 	glass_name = "glass of coffee"
 	glass_desc = "Don't drop it, or you'll send scalding liquid and glass shards everywhere."
 
 /datum/reagent/consumable/coffee/overdose_process(mob/living/M)
-	M.Jitter(5)
+	var/datum/disease/D = new /datum/disease/heart_failure
```
Comment (Ghommie) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]:
```suggestion
	var/datum/disease/heart_failure/D = /datum/disease/heart_failure
```
Don't waste effort creating a new heart failure disease everytime if the check below won't pass (typepaths work just fine for it).

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm [hunk 2]
```diff
@@ -262,20 +262,61 @@
...
+	if(!M.HasDisease(D))
+		M.ForceContractDisease(D)
```
Comment (Ghommie) [inline code/modules/reagents/chemistry/reagents/drink_reagents.dm]:
```suggestion
		M.ForceContractDisease(new D)
```

**Review by Ghommie — COMMENTED** [review]
The heart failure event on overdose process should only have a probability of 1 to 5, instead of triggering tick.
Take a look at the thirteen loko overdose for reference.

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 1 above)
Comment (Thalpy) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm]: Use a check for current cycle if you want to run a limited bit of code, or use overdose_start() instead

Comment (Ghommie) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm]: overdose_start() is only called one time, you can cheese it with just a little bit of corazone afterward.

Comment (Thalpy) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm]: You want it to constantly apply the status? What if someone treats you with cora and assumes you're fine, or assumes it's bugged?
(I'm just curious and not attacking you..!)

**Comment by Tupinambis** [comment]
Heart failure change good. Stamina decrease change is bad, and I just see there being a coffee meta where everyone has GOT to be carrying a cup of coffee at all times.

File: code/modules/reagents/chemistry/reagents/drink_reagents.dm (hunk 1 above)
Comment (Sishen1542) [reply code/modules/reagents/chemistry/reagents/drink_reagents.dm]: Constantly applying while overdosing seems like a good idea, as the heart will be in constant overdrive. Will apply the above changes and see how it works out though!

**Comment by Sishen1542** [comment]
> The heart failure event on overdose process should only have a probability of 1 to 5, instead of triggering tick.
> Take a look at the thirteen loko overdose for reference.

I looked at thirteen loko originally for it, yeah. The only issue is that I wanted harsh punishments since restoring stamina is a POWERFUL tool.

**Comment by silicons** [comment]
no thank you stamina man is not so good right now but adding a very easily gotten stamina restorer is far worse.

**Pull request closed without merging**